t/t2605-job-shell-output-redirection-standalone.t

```diff
@@ -205,25 +207,60 @@ test_expect_success HAVE_JQ 'flux-shell: run 2-task echo job (stdout term/stderr
 #
 
 test_expect_success HAVE_JQ 'flux-shell: run 1-task echo job (mustache id stdout file/stderr file)' '
+    id=$(flux job id --to=f58 14) &&
         cat j1echoboth \
             |  $jq ".attributes.system.shell.options.output.stdout.type = \"file\"" \
-            |  $jq ".attributes.system.shell.options.output.stdout.path = \"out{{id}}\"" \
+            |  $jq ".attributes.system.shell.options.output.stdout.path = \"out{{jobid}}\"" \
             |  $jq ".attributes.system.shell.options.output.stderr.type = \"file\"" \
-            |  $jq ".attributes.system.shell.options.output.stderr.path = \"err{{id}}\"" \
+            |  $jq ".attributes.system.shell.options.output.stderr.path = \"err{{jobid}}\"" \
             > j1echoboth-14 &&
 	${FLUX_SHELL} -v -s -r 0 -j j1echoboth-14 -R R1 14 &&
-	grep stdout:baz out14 &&
-	grep stderr:baz err14
+	grep stdout:baz out${id} &&
+	grep stderr:baz err${id}
 '
 
 test_expect_success HAVE_JQ 'flux-shell: run 1-task echo job (mustache id stdout & stderr to stdout file)' '
+    id=$(flux job id --to=f58 15) &&
         cat j1echoboth \
             |  $jq ".attributes.system.shell.options.output.stdout.type = \"file\"" \
             |  $jq ".attributes.system.shell.options.output.stdout.path = \"out{{id}}\"" \
             > j1echoboth-15 &&
 	${FLUX_SHELL} -v -s -r 0 -j j1echoboth-15 -R R1 15 &&
-	grep stdout:baz out15 &&
-	grep stderr:baz out15
+	grep stdout:baz out${id} &&
+	grep stderr:baz out${id}
+'
+
+for type in f58 dec hex dothex words; do
+  test_expect_success HAVE_JQ "flux-shell: output template {{id.$type}}" '
+    jobid=123456789 &&
+    id=$(flux job id --to=${type} ${jobid}) &&
+        cat j1echoboth \
+            |  $jq ".attributes.system.shell.options.output.stdout.type = \"file\"" \
+            |  $jq ".attributes.system.shell.options.output.stdout.path = \"out{{id.${type}}}\"" \
+            > j1-mustache-${type} &&
+	${FLUX_SHELL} -v -s -r 0 -j j1-mustache-${type} -R R1 ${jobid} &&
+	grep stdout:baz out${id} &&
+	grep stderr:baz out${id}
+ '
+done
+
+test_expect_success HAVE_JQ "flux-shell: bad output mustache template is not rendered" '
+	cat j1echoboth \
+	    |  $jq ".attributes.system.shell.options.output.stdout.type = \"file\"" \
+	    |  $jq ".attributes.system.shell.options.output.stdout.path = \"{{idx}}.out\"" \
+	    > j1-mustache-error1 &&
+	${FLUX_SHELL} -v -s -r 0 -j j1-mustache-error1 -R R1 1234 &&
+	grep stdout:baz {{idx}}.out &&
+	grep stderr:baz {{idx}}.out
+'
+test_expect_success HAVE_JQ "flux-shell: bad output mustache template is not rendered" '
+	cat j1echoboth \
+	    |  $jq ".attributes.system.shell.options.output.stdout.type = \"file\"" \
+	    |  $jq ".attributes.system.shell.options.output.stdout.path = \"{{id.x}}.out\"" \
+	    > j1-mustache-error2 &&
+	${FLUX_SHELL} -v -s -r 0 -j j1-mustache-error2 -R R1 1234 &&
+	grep stdout:baz {{id.x}}.out &&
+	grep stderr:baz {{id.x}}.out
```

Comment:
since `mustache_renderer()` loops and allows multiple mustache templates in a single output filename, perhaps some double mustache tests would be good, like `job-{{id}{{jobid}}-{{id.f58}}.out` or something.

Reply:
Good idea.